packages/wdio-browserstack-service/src/launcher.ts

```diff
@@ -291,20 +333,124 @@ export default class BrowserstackLauncherService implements Services.ServiceInst
+                        }
+                    } else if (capType === 'localIdentifier') {
+                        (caps.capabilities as Capabilities.Capabilities)['browserstack.localIdentifier'] = value
                     }
                 } else if (capType === 'local'){
                     (caps.capabilities as Capabilities.Capabilities)['bstack:options']!.local = true
                 } else if (capType === 'app') {
                     (caps.capabilities as Capabilities.Capabilities)['appium:app'] = value
+                } else if (capType === 'buildIdentifier') {
+                    if (value) {
+                        (caps.capabilities as Capabilities.Capabilities)['bstack:options']!.buildIdentifier = value
+                    } else {
+                        delete (caps.capabilities as Capabilities.Capabilities)['bstack:options']!.buildIdentifier
+                    }
+                } else if (capType === 'localIdentifier') {
+                    (caps.capabilities as Capabilities.Capabilities)['bstack:options']!.localIdentifier = value
                 }
             })
         } else {
             throw new SevereServiceError('Capabilities should be an object or Array!')
         }
     }
+
+    _handleBuildIdentifier(capabilities?: Capabilities.RemoteCapabilities) {
+        if (!this._buildIdentifier) {
+            this._updateCaps(capabilities, 'buildIdentifier')
+            return
+        }
+
+        if (!this._buildName || process.env.BROWSERSTACK_BUILD_NAME) {
+            if (this._buildIdentifier) {
+                this._updateCaps(capabilities, 'buildIdentifier')
+                log.warn('Skipping buildIdentifier as buildName is not passed.')
+                return
+            }
+        }
+
+        if (this._buildIdentifier && this._buildIdentifier.includes('${DATE_TIME}')){
+            const formattedDate = new Intl.DateTimeFormat('en-GB', {
+                month: 'short',
+                day: '2-digit',
+                hour: '2-digit',
+                minute: '2-digit',
+                hour12: false })
+                .format(new Date())
+                .replace(/ |, /g, '-')
+            this._buildIdentifier = this._buildIdentifier.replace('${DATE_TIME}', formattedDate)
+            this._updateCaps(capabilities, 'buildIdentifier', this._buildIdentifier)
+        }
+
+        if (!this._buildIdentifier.includes('${BUILD_NUMBER}')) return
+
+        const ciInfo = getCiInfo()
+        if (ciInfo != null) {
+            const ciBuildNumber = ciInfo.build_number
+            if (ciBuildNumber != null) {
+                this._buildIdentifier = this._buildIdentifier.replace('${BUILD_NUMBER}', 'CI '+ ciBuildNumber)
+                this._updateCaps(capabilities, 'buildIdentifier', this._buildIdentifier)
+            }
+        } else {
+            const localBuildNumber = this._getLocalBuildNumber()
+            if (localBuildNumber != '-1') {
```

Comment:
Why returning `-1` / `1` ? IMO it is better work with `true` and `false` here and call variable `hasCreatedLocalBuildSuccessfully`

Reply:
Thank you for the review. 
We're returning value as `-1` and `1` because we need to return `1` in case buildName doesn't exist in json file. So can't return a boolean in that case

Reply:
From what I am reading `_getLocalBuildNumber` returns `1` in case of success and `-1` in case of failure, why can't we use `true` or `false` for this?

Reply:
No.
Suppose `buildName` is `wdio-build`, if `buildName` doesn't exists in json file then it will return `1` else returns corresponding value in json file (e.g. {"wdio-build":{"identifier":2}} => `2` in this case)

Reply:
I see, can we document this as comment above the `_getLocalBuildNumber` function? Also how about we return `undefined` in case of an error?

Reply:
Good suggestion, I can document this as comment. What if I return `null` instead of `-1` or `undefined` ? would be more relevant, signifies no value returned.

Reply:
@christian-bromann Have incorporated the comment, please have a look. I think all comments/suggestions are resolved in this PR.